crates/amaru-kernel/src/lib.rs

```diff
@@ -533,6 +538,54 @@ pub fn to_ex_units(witness_set: WitnessSet) -> ExUnits {
     }
 }
 
+pub trait HasAddress {
+    fn address(&self) -> Result<Address, pallas_addresses::Error>;
+}
+
+impl HasAddress for TransactionOutput {
+    fn address(&self) -> Result<Address, pallas_addresses::Error> {
+        match self {
+            PseudoTransactionOutput::Legacy(transaction_output) => {
+                Address::from_bytes(&transaction_output.address)
+            }
+            PseudoTransactionOutput::PostAlonzo(modern) => Address::from_bytes(&modern.address),
+        }
+    }
+}
+
+pub trait HasKeyHash {
+    fn get_key_hash(&self) -> Option<Hash<28>>;
```

Comment:
nit: maybe this should be named `key_hash`? since that matches the naming conventions of pallas, and of the other helper trait in here